cloud/scope/machine.go

```diff
@@ -328,10 +328,19 @@ func (m *MachineScope) ProviderID() string {
 
 // AvailabilitySet returns the availability set for this machine if available
 func (m *MachineScope) AvailabilitySet() (string, bool) {
-	if m.IsControlPlane() && m.AvailabilitySetEnabled() {
+	if len(m.FailureDomains()) > 0 {
```

Comment:
I think I prefer the explicit `AvailabilitySetEnabled()` and hiding the details of what that means (`len(s.AzureCluster.Status.FailureDomains) == 0`) in helper func

Reply:
at the very least we should add a comment to explain that why don't define an availability set when there are failure domains available

Reply:
Yeah, Agreed. Fixed it.